packages/config/src/index.ts

```diff
@@ -186,12 +186,25 @@ export default async (
     'workspace-concurrency': 4,
     'workspace-prefix': opts.workspaceDir,
   })
+
+  const settingKeys = Object.keys(npmConfig?.sources?.project?.data ?? {})
```

Comment:
Might be good to also check the setting file that is at the root of the workspace. That's probably `npmConfig?.sources?.workspace`

and the user config as well `npmConfig?.sources?.user`